packages/dashboard/src/components/confirm-alert-dialog.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Slide from '@material-ui/core/Slide';
+import { TransitionProps } from '@material-ui/core/transitions';
+import WarningIcon from '@material-ui/icons/Warning';
+import { makeStyles } from '@material-ui/core';
+
+const Transition = React.forwardRef(function Transition(
+  props: TransitionProps & { children?: React.ReactElement<any, any> },
+  ref: React.Ref<unknown>,
+) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+export interface ConfirmAlertDialogProps {
+  open: boolean; // controls when the modal should open
+  close: () => void; // close the modal
+  title?: string;
+  content?: string;
+  showIcon?: boolean;
+  confirmButtonText?: string;
+  cancelButtonText?: string;
+  confirmCallback?: () => void;
+  cancelCallback?: () => void;
+}
+
+/**
+ * Modal to warn a user about something related to an action.
+ */
+export const ConfirmAlertDialog = (props: ConfirmAlertDialogProps) => {
+  const {
+    open,
+    close,
+    title,
+    content,
+    showIcon,
+    confirmButtonText,
+    cancelButtonText,
+    confirmCallback,
+    cancelCallback,
+  } = props;
+
+  const handleCancelButton = () => {
+    cancelCallback && cancelCallback();
+    close();
+  };
+
+  const handleConfirmButton = () => {
+    confirmCallback && confirmCallback();
+    close();
+  };
+  const classes = useStyles();
+
+  return (
+    <div>
+      <Dialog
+        open={open}
+        TransitionComponent={Transition}
+        keepMounted
```

Comment:
We should let the parent component decides if it wants to keep it mounted.

Reply:
Makes sense!